test/features/financials/ReturnsServiceSpec.scala

```diff
@@ -0,0 +1,211 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import java.time.LocalDate
+
+import connectors.RegistrationConnector
+import features.financials.models.{Frequency, Returns, Stagger}
+import helpers.VatRegSpec
+import models.S4LKey
+import org.mockito.ArgumentMatchers.any
+import org.scalatest.MustMatchers
+import org.scalatest.mockito.MockitoSugar
+import org.mockito.Mockito._
+import play.api.libs.json.JsString
+import uk.gov.hmrc.http.cache.client.CacheMap
+import uk.gov.hmrc.http.{HttpResponse, NotFoundException}
+
+import scala.concurrent.Future
+
+
+class ReturnsServiceSpec extends VatRegSpec with MustMatchers with MockitoSugar {
+
+  class Setup {
+    val service = new ReturnsService {
+      override val s4lService: S4LService = mockS4LService
+      override val vatService: RegistrationService = mockVatRegistrationService
+      override val vatRegConnector: RegistrationConnector = mockRegConnector
+    }
+  }
+
+  val mockCacheMap = CacheMap("", Map("" -> JsString("")))
+
+  val date         = LocalDate.now
+  val returns      = Returns(Some(true), Some(Frequency.quarterly), Some(Stagger.apr_may_jun), Some(date))
+  val emptyReturns = Returns(None, None, None, None)
+  val incomplete   = emptyReturns.copy(reclaimVatOnMostReturns = Some(true))
+
+  "getReturnsViewModel" should {
+    "return a model from Save4Later" in new Setup {
+      when(mockS4LService.fetchAndGetNoAux[Returns](any[S4LKey[Returns]]())(any(), any(), any()))
+        .thenReturn(Future.successful(Some(returns)))
+
+      await(service.getReturns) mustBe returns
+    }
+
+    "return a model from MongoDB" in new Setup {
+      when(mockS4LService.fetchAndGetNoAux[Returns](any[S4LKey[Returns]]())(any(), any(), any()))
+        .thenReturn(Future.successful(None))
+      when(mockRegConnector.getReturns(any())(any(), any()))
+        .thenReturn(Future.successful(returns))
+
+      await(service.getReturns) mustBe returns
+    }
+
+    "construct a blank model when nothing was found in Save4Later or Mongo" in  new Setup {
+      when(mockS4LService.fetchAndGetNoAux[Returns](any[S4LKey[Returns]]())(any(), any(), any()))
+        .thenReturn(Future.successful(None))
+      when(mockRegConnector.getReturns(any())(any(), any()))
+        .thenReturn(Future.failed(new NotFoundException("404")))
+
+      await(service.getReturns) mustBe emptyReturns
+    }
+    "construct a blank model when an exception occurred fetching from Save4Later or Mongo" in new Setup {
+      when(mockS4LService.fetchAndGetNoAux[Returns](any[S4LKey[Returns]]())(any(), any(), any()))
+        .thenReturn(Future.failed(new RuntimeException("BOOM")))
+
+      await(service.getReturns) mustBe emptyReturns
+    }
+  }
+
+  "handleView" should {
+    "should return a Complete frequency is decided for the user" in new Setup {
```

Comment:
Was this test supposed to be using defaultedReturns? as currently it is testing the same as the test below it.